docs/how-to/cooperative_groups.rst

```diff
@@ -0,0 +1,468 @@
+.. meta::
+  :description: This topic describes how to use cooperative groups in HIP
+  :keywords: AMD, ROCm, HIP, cooperative groups
+
+.. _cooperative_groups_how-to:
+
+*******************************************************************************
+Cooperative Groups
+*******************************************************************************
+
+Cooperative groups API is an extension to the HIP programming model, which provides developers with a flexible, dynamic grouping mechanism for the communicating threads. Cooperative Groups let you define your own set of thread groups which may fit your user-cases better than those defined by the hardware. This lets you specify the level of granularity for thread communication which can lead to more efficient parallel decompositions.
+
+The API is accessible in the ``cooperative_groups`` namespace after the  ``hip_cooperative_groups.h`` is included. The header contains the following  elements:
+
+* Static functions to create groups and subgroups.
+* Hardware-accelerated operations over the whole group, like shuffles.
+* Data types of cooperative groups.
+* Synchronize member function of the groups.
+* Get group properties member functions.
+
+Cooperative groups thread model
+===============================
+
+The thread hierarchy abstraction of cooperative groups is in
+:numref:`coop_thread_hierarchy`.
+
+.. _coop_thread_hierarchy:
+
+.. figure:: ../data/how-to/cooperative_groups/thread_hierarchy_coop.svg
+  :alt: Diagram depicting nested rectangles of varying color. The outermost one
+        titled "Grid", inside sets of different sized rectangles layered on
+        one another titled "Block". Each "Block" containing sets of uniform
+        rectangles layered on one another titled "Warp". Each of the "Warp"
+        titled rectangles filled with downward pointing arrows inside.
+
+  Cooperative group thread hierarchy.
```

Comment:
This isn't really the entire hierarchy.  Maybe call it "top of cooperative groups thread hierarchy"?